rs/registry/canister/src/mutations/node_management/do_add_node.rs

```diff
@@ -585,15 +587,17 @@ mod tests {
         let mut registry = invariant_compliant_registry(0);
         // Add node operator record first
         let node_operator_record = NodeOperatorRecord {
-            max_rewardable_nodes: btreemap! { "type1".to_string() => 28 },
+            max_rewardable_nodes: btreemap! { "type1".to_string() => 1 },
             ..Default::default()
         };
         let node_operator_id = PrincipalId::from_str(TEST_NODE_ID).unwrap();
         registry.maybe_apply_mutation_internal(vec![insert(
             make_node_operator_record_key(node_operator_id),
             node_operator_record.encode_to_vec(),
         )]);
-        let (payload, _) = prepare_add_node_payload(1, "type1");
+        let (mut payload, _) = prepare_add_node_payload(1);
+        payload.node_reward_type = Some(NodeRewardType::Type1.to_string());
```

Comment:
No change required.

It seems like you generally set node_reward_type right after calling prepare_add_node_payload. Maybe, node_reward_type should be an argumnet?